File: 20-Lambdas-and-Built-In-Functions/71-extract_full_name.py
# == using map ===
def extract_full_name(names):
    return list(map(lambda name: f"{name['first']} {name['last']}"), names)

# === using list comprehension ===
def extract_full_name(names):
    return [' '.join([name['first'], name['last']]) for name in names]


# === tests ===
names = [{'first': 'Elie', 'last': 'Schoppik'}, {'first': 'Colt', 'last': 'Steele'}]
print(extract_full_name(names))

File: 20-Lambdas-and-Built-In-Functions/71-extract_full_name.js
function extractFullName(names) {
    return names.map(name => `${name['first']} ${name['last']}`)
}


// === tests ===
let names = [{'first': 'Elie', 'last': 'Schoppik'}, {'first': 'Colt', 'last': 'Steele'}]
console.log(extractFullName(names))